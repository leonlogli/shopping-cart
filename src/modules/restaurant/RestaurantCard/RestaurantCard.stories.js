import React from 'react'

import RestaurantCard from '.'

export default {
  title: 'RestaurantCard',
}

export const Default = () => (
  <RestaurantCard
    title="Restaurant Principal"
    image="https://www.uir.ac.ma/upload/cbuilder/3e5aa8eb56fb64e0d9340fa21687e7693c14fe56.png"
    category="Repas variés"
  />
)

export const closed = () => (
  <RestaurantCard
    title="Restaurant Principal"
    image="https://www.uir.ac.ma/upload/cbuilder/3e5aa8eb56fb64e0d9340fa21687e7693c14fe56.png"
    category="Repas variés"
    open={false}
  />
)
