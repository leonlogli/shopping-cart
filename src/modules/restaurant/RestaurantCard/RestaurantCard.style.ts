import styled, { css } from 'styled-components'
import CardActionArea from '@material-ui/core/CardActionArea'

import { Text, H6, Box } from '../../../components'

const Info = styled(Box)`
  position: absolute;
  bottom: 8px;
  left: 8px;
  background: rgba(0, 0, 0, 0.8);
  color: gainsboro;
  padding: 8px 12px;
  border-radius: 25px;
`

const Title = styled(H6)`
  margin: 8px auto;
  margin-top: 4.5%;
  width: calc(100% - 24px);
  text-align: center;
  padding: 12px;
  box-shadow: ${({ theme }) => theme.shadows[1]};
  font-size: 14px;
  background: ${({ theme }) => theme.palette.background.paper};
  text-transform: uppercase;
  border-radius: 25px;
  max-width: 400px;
`

const ContentBox = styled(CardActionArea)`
  display: flex;
  align-items: flex-start;
  position: relative;
  box-shadow: ${({ theme }) => theme.shadows[1]};
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  flex-grow: 1;
  border-radius: 25px;
  margin-bottom: 8px;

  ${({ image }: any) => css`
    background-image: url(${image});
  `}
`

const StatusText = styled(Text)`
  font-size: 15px;
  font-weight: 600;
  text-align: center;

  ${({ open, theme }: any) => css`
    color: ${!open ? theme.palette.error.main : theme.palette.primary.main};
  `}
`

const RestaurantCardRoot = styled(Box)`
  border-radius: 25px;
  min-width: 200px;
`

export { Title, ContentBox, RestaurantCardRoot, Info, StatusText }
