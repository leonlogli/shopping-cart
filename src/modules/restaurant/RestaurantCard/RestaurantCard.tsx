import React from 'react'

import { RatioBox, BoxProps } from '../../../components'

import {
  ContentBox,
  RestaurantCardRoot,
  Info,
  Title,
  StatusText,
} from './RestaurantCard.style'

export type RestaurantCardProps = BoxProps & {
  title: string
  category: string
  image: string
  open?: boolean
  status?: string
}

const RestaurantCard = (props: RestaurantCardProps) => {
  const {
    title,
    category,
    open = true,
    status,
    image,
    className,
    ...other
  } = props
  const classNameProp = `RestaurantCard ${className || ''}`

  return (
    <RestaurantCardRoot className={classNameProp} {...other}>
      <RatioBox ratio={80}>
        <ContentBox {...{ image }}>
          <Title className="Title">{title}</Title>
          <Info>{category}</Info>
        </ContentBox>
        <StatusText {...{ open }} variant="button">
          {status || open ? 'open' : 'closed'}
        </StatusText>
      </RatioBox>
    </RestaurantCardRoot>
  )
}

export { RestaurantCard }
export default RestaurantCard
