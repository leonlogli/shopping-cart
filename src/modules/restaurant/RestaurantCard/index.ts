export * from './RestaurantCard'
export { default } from './RestaurantCard'
