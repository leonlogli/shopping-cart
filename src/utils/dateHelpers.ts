const durationFormat = (interval: number, type: string) => {
  return `il y a ${Math.floor(interval)} ${type}${
    Math.floor(interval) > 1 ? 's' : ''
  }`
}

const fromNow = (date: Date) => {
  const seconds = Math.floor(
    ((new Date() as any) - (new Date(date) as any)) / 1000
  )

  let interval = seconds / 31536000

  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'an')
  }

  interval = seconds / 2592000
  if (Math.floor(interval) >= 1) {
    return `il y a ${Math.floor(interval)} mois`
  }

  interval = seconds / 86400
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'jour')
  }

  interval = seconds / 3600
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'heure')
  }

  interval = seconds / 60
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'minute')
  }

  return `Ã  l'instant`
}

/**
 * Parse the given string time to a date.Time must have `hh:mm:ss` format
 * @param time time to parse
 * @param date date to which the time will be set once parsed. If null, today's date is used
 */
const parseTimeToDate = (time: string, date?: Date) => {
  const dateToSet = date || new Date()
  const timeParts = time.split(':').map((t) => Number(t) || 0)

  dateToSet.setHours(timeParts[0], timeParts[1], timeParts[2] || 0)

  return dateToSet
}

const formatShortDate = (date: any, locale?: string) => {
  return new Date(date).toLocaleDateString(locale, {
    day: 'numeric',
    month: 'short',
    year: 'numeric',
  })
}

export { fromNow, parseTimeToDate, formatShortDate }
