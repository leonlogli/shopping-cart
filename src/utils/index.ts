export * from './testHelpers'
export * from './util'
export * from './dateHelpers'
