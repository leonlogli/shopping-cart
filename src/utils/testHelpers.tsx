import React from 'react'
import { ThemeProvider } from 'styled-components'
import { mount, shallow } from 'enzyme'
import { defaultTheme } from '../styles'

const ThemeProviderWrapper = ({ children }: any) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
)

export const shallowWithTheme = (tree: any) =>
  shallow(tree, {
    wrappingComponent: ThemeProviderWrapper,
  })

export const mountWithTheme = (tree: any) =>
  mount(tree, {
    wrappingComponent: ThemeProviderWrapper,
  })
