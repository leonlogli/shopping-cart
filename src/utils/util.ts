const toUSD = (value: number) => {
  return `${value.toLocaleString('en-US', {
    minimumFractionDigits: 2,
  })} USD`
}

export { toUSD }
