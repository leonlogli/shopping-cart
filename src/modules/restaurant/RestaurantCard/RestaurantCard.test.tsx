import React from 'react'

import { mountWithTheme } from '../../../utils'

import RestaurantCard from './RestaurantCard'

describe('<RestaurantCard />', () => {
  const wrapper = mountWithTheme(
    <RestaurantCard
      title="Restaurant Principal"
      image="https://www.uir.ac.ma/upload/cbuilder/3e5aa8eb56fb64e0d9340fa21687e7693c14fe56.png"
      category="Repas variÃ©s"
    />
  )

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
