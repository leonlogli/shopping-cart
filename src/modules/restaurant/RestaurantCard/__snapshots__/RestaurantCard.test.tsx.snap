// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RestaurantCard /> should match a snapshot 1`] = `
<RestaurantCard
  category="Repas variÃ©s"
  image="https://www.uir.ac.ma/upload/cbuilder/3e5aa8eb56fb64e0d9340fa21687e7693c14fe56.png"
  title="Restaurant Principal"
>
  <Styled(Component)
    className="RestaurantCard "
  >
    <ForwardRef
      className="euvjqF RestaurantCard "
    >
      <Styled(Styled(MuiBox))
        className="euvjqF RestaurantCard "
      >
        <Styled(MuiBox)
          className="dUNeVP euvjqF RestaurantCard "
        >
          <div
            className="MuiBox-root MuiBox-root-1 dUNeVP euvjqF RestaurantCard "
          >
            <RatioBox
              ratio={94}
            >
              <ForwardRef>
                <Styled(Styled(MuiBox))>
                  <Styled(MuiBox)
                    className="dUNeVP"
                  >
                    <div
                      className="MuiBox-root MuiBox-root-2 dUNeVP"
                    >
                      <Styled(Component)
                        ratio={94}
                      >
                        <ForwardRef
                          className="fxmcfe"
                          ratio={94}
                        >
                          <Styled(Styled(MuiBox))
                            className="fxmcfe"
                            ratio={94}
                          >
                            <Styled(MuiBox)
                              className="dUNeVP fxmcfe"
                              ratio={94}
                            >
                              <div
                                className="MuiBox-root MuiBox-root-3 dUNeVP fxmcfe"
                                ratio={94}
                              >
                                <Styled(Component)>
                                  <ForwardRef
                                    className="kxbNdS"
                                  >
                                    <Styled(Styled(MuiBox))
                                      className="kxbNdS"
                                    >
                                      <Styled(MuiBox)
                                        className="dUNeVP kxbNdS"
                                      >
                                        <div
                                          className="MuiBox-root MuiBox-root-4 dUNeVP kxbNdS"
                                        >
                                          <Styled(WithStyles(ForwardRef(CardActionArea)))
                                            image="https://www.uir.ac.ma/upload/cbuilder/3e5aa8eb56fb64e0d9340fa21687e7693c14fe56.png"
                                          >
                                            <WithStyles(ForwardRef(CardActionArea))
                                              className="kgIEMH"
                                              image="https://www.uir.ac.ma/upload/cbuilder/3e5aa8eb56fb64e0d9340fa21687e7693c14fe56.png"
                                            >
                                              <ForwardRef(CardActionArea)
                                                className="kgIEMH"
                                                classes={
                                                  Object {
                                                    "focusHighlight": "MuiCardActionArea-focusHighlight",
                                                    "focusVisible": "Mui-focusVisible",
                                                    "root": "MuiCardActionArea-root",
                                                  }
                                                }
                                                image="https://www.uir.ac.ma/upload/cbuilder/3e5aa8eb56fb64e0d9340fa21687e7693c14fe56.png"
                                              >
                                                <WithStyles(ForwardRef(ButtonBase))
                                                  className="MuiCardActionArea-root kgIEMH"
                                                  focusVisibleClassName="Mui-focusVisible"
                                                  image="https://www.uir.ac.ma/upload/cbuilder/3e5aa8eb56fb64e0d9340fa21687e7693c14fe56.png"
                                                >
                                                  <ForwardRef(ButtonBase)
                                                    className="MuiCardActionArea-root kgIEMH"
                                                    classes={
                                                      Object {
                                                        "disabled": "Mui-disabled",
                                                        "focusVisible": "Mui-focusVisible",
                                                        "root": "MuiButtonBase-root",
                                                      }
                                                    }
                                                    focusVisibleClassName="Mui-focusVisible"
                                                    image="https://www.uir.ac.ma/upload/cbuilder/3e5aa8eb56fb64e0d9340fa21687e7693c14fe56.png"
                                                  >
                                                    <button
                                                      className="MuiButtonBase-root MuiCardActionArea-root kgIEMH"
                                                      disabled={false}
                                                      image="https://www.uir.ac.ma/upload/cbuilder/3e5aa8eb56fb64e0d9340fa21687e7693c14fe56.png"
                                                      onBlur={[Function]}
                                                      onDragLeave={[Function]}
                                                      onFocus={[Function]}
                                                      onKeyDown={[Function]}
                                                      onKeyUp={[Function]}
                                                      onMouseDown={[Function]}
                                                      onMouseLeave={[Function]}
                                                      onMouseUp={[Function]}
                                                      onTouchEnd={[Function]}
                                                      onTouchMove={[Function]}
                                                      onTouchStart={[Function]}
                                                      tabIndex={0}
                                                      type="button"
                                                    >
                                                      <Styled(H6)
                                                        className="Title"
                                                      >
                                                        <H6
                                                          className="fOaJtt Title"
                                                        >
                                                          <Styled(WithStyles(ForwardRef(Typography)))
                                                            className="fOaJtt Title"
                                                            variant="h6"
                                                          >
                                                            <WithStyles(ForwardRef(Typography))
                                                              className="fOaJtt Title"
                                                              variant="h6"
                                                            >
                                                              <ForwardRef(Typography)
                                                                className="fOaJtt Title"
                                                                classes={
                                                                  Object {
                                                                    "alignCenter": "MuiTypography-alignCenter",
                                                                    "alignJustify": "MuiTypography-alignJustify",
                                                                    "alignLeft": "MuiTypography-alignLeft",
                                                                    "alignRight": "MuiTypography-alignRight",
                                                                    "body1": "MuiTypography-body1",
                                                                    "body2": "MuiTypography-body2",
                                                                    "button": "MuiTypography-button",
                                                                    "caption": "MuiTypography-caption",
                                                                    "colorError": "MuiTypography-colorError",
                                                                    "colorInherit": "MuiTypography-colorInherit",
                                                                    "colorPrimary": "MuiTypography-colorPrimary",
                                                                    "colorSecondary": "MuiTypography-colorSecondary",
                                                                    "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                                                    "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                                                    "displayBlock": "MuiTypography-displayBlock",
                                                                    "displayInline": "MuiTypography-displayInline",
                                                                    "gutterBottom": "MuiTypography-gutterBottom",
                                                                    "h1": "MuiTypography-h1",
                                                                    "h2": "MuiTypography-h2",
                                                                    "h3": "MuiTypography-h3",
                                                                    "h4": "MuiTypography-h4",
                                                                    "h5": "MuiTypography-h5",
                                                                    "h6": "MuiTypography-h6",
                                                                    "noWrap": "MuiTypography-noWrap",
                                                                    "overline": "MuiTypography-overline",
                                                                    "paragraph": "MuiTypography-paragraph",
                                                                    "root": "MuiTypography-root",
                                                                    "srOnly": "MuiTypography-srOnly",
                                                                    "subtitle1": "MuiTypography-subtitle1",
                                                                    "subtitle2": "MuiTypography-subtitle2",
                                                                  }
                                                                }
                                                                variant="h6"
                                                              >
                                                                <h6
                                                                  className="MuiTypography-root fOaJtt Title MuiTypography-h6"
                                                                >
                                                                  Restaurant Principal
                                                                </h6>
                                                              </ForwardRef(Typography)>
                                                            </WithStyles(ForwardRef(Typography))>
                                                          </Styled(WithStyles(ForwardRef(Typography)))>
                                                        </H6>
                                                      </Styled(H6)>
                                                      <Styled(Component)>
                                                        <ForwardRef
                                                          className="hueOnz"
                                                        >
                                                          <Styled(Styled(MuiBox))
                                                            className="hueOnz"
                                                          >
                                                            <Styled(MuiBox)
                                                              className="dUNeVP hueOnz"
                                                            >
                                                              <div
                                                                className="MuiBox-root MuiBox-root-5 dUNeVP hueOnz"
                                                              >
                                                                Repas variÃ©s
                                                              </div>
                                                            </Styled(MuiBox)>
                                                          </Styled(Styled(MuiBox))>
                                                        </ForwardRef>
                                                      </Styled(Component)>
                                                      <span
                                                        className="MuiCardActionArea-focusHighlight"
                                                      />
                                                      <WithStyles(memo)
                                                        center={false}
                                                      >
                                                        <ForwardRef(TouchRipple)
                                                          center={false}
                                                          classes={
                                                            Object {
                                                              "child": "MuiTouchRipple-child",
                                                              "childLeaving": "MuiTouchRipple-childLeaving",
                                                              "childPulsate": "MuiTouchRipple-childPulsate",
                                                              "ripple": "MuiTouchRipple-ripple",
                                                              "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                                                              "rippleVisible": "MuiTouchRipple-rippleVisible",
                                                              "root": "MuiTouchRipple-root",
                                                            }
                                                          }
                                                        >
                                                          <span
                                                            className="MuiTouchRipple-root"
                                                          >
                                                            <TransitionGroup
                                                              childFactory={[Function]}
                                                              component={null}
                                                              exit={true}
                                                            />
                                                          </span>
                                                        </ForwardRef(TouchRipple)>
                                                      </WithStyles(memo)>
                                                    </button>
                                                  </ForwardRef(ButtonBase)>
                                                </WithStyles(ForwardRef(ButtonBase))>
                                              </ForwardRef(CardActionArea)>
                                            </WithStyles(ForwardRef(CardActionArea))>
                                          </Styled(WithStyles(ForwardRef(CardActionArea)))>
                                          <Styled(Text)
                                            open={true}
                                            variant="button"
                                          >
                                            <Text
                                              className="gUpNge"
                                              open={true}
                                              variant="button"
                                            >
                                              <Styled(WithStyles(ForwardRef(Typography)))
                                                className="gUpNge"
                                                open={true}
                                                variant="button"
                                              >
                                                <WithStyles(ForwardRef(Typography))
                                                  className="gUpNge"
                                                  open={true}
                                                  variant="button"
                                                >
                                                  <ForwardRef(Typography)
                                                    className="gUpNge"
                                                    classes={
                                                      Object {
                                                        "alignCenter": "MuiTypography-alignCenter",
                                                        "alignJustify": "MuiTypography-alignJustify",
                                                        "alignLeft": "MuiTypography-alignLeft",
                                                        "alignRight": "MuiTypography-alignRight",
                                                        "body1": "MuiTypography-body1",
                                                        "body2": "MuiTypography-body2",
                                                        "button": "MuiTypography-button",
                                                        "caption": "MuiTypography-caption",
                                                        "colorError": "MuiTypography-colorError",
                                                        "colorInherit": "MuiTypography-colorInherit",
                                                        "colorPrimary": "MuiTypography-colorPrimary",
                                                        "colorSecondary": "MuiTypography-colorSecondary",
                                                        "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                                        "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                                        "displayBlock": "MuiTypography-displayBlock",
                                                        "displayInline": "MuiTypography-displayInline",
                                                        "gutterBottom": "MuiTypography-gutterBottom",
                                                        "h1": "MuiTypography-h1",
                                                        "h2": "MuiTypography-h2",
                                                        "h3": "MuiTypography-h3",
                                                        "h4": "MuiTypography-h4",
                                                        "h5": "MuiTypography-h5",
                                                        "h6": "MuiTypography-h6",
                                                        "noWrap": "MuiTypography-noWrap",
                                                        "overline": "MuiTypography-overline",
                                                        "paragraph": "MuiTypography-paragraph",
                                                        "root": "MuiTypography-root",
                                                        "srOnly": "MuiTypography-srOnly",
                                                        "subtitle1": "MuiTypography-subtitle1",
                                                        "subtitle2": "MuiTypography-subtitle2",
                                                      }
                                                    }
                                                    open={true}
                                                    variant="button"
                                                  >
                                                    <span
                                                      className="MuiTypography-root gUpNge MuiTypography-button"
                                                      open={true}
                                                    >
                                                      open
                                                    </span>
                                                  </ForwardRef(Typography)>
                                                </WithStyles(ForwardRef(Typography))>
                                              </Styled(WithStyles(ForwardRef(Typography)))>
                                            </Text>
                                          </Styled(Text)>
                                        </div>
                                      </Styled(MuiBox)>
                                    </Styled(Styled(MuiBox))>
                                  </ForwardRef>
                                </Styled(Component)>
                              </div>
                            </Styled(MuiBox)>
                          </Styled(Styled(MuiBox))>
                        </ForwardRef>
                      </Styled(Component)>
                    </div>
                  </Styled(MuiBox)>
                </Styled(Styled(MuiBox))>
              </ForwardRef>
            </RatioBox>
          </div>
        </Styled(MuiBox)>
      </Styled(Styled(MuiBox))>
    </ForwardRef>
  </Styled(Component)>
</RestaurantCard>
`;
